### ISSUE TITLE
Time stepping

### ISSUE THREAD

**Issue by koehlerson** [KnutAM/FESolvers.jl#6]
We currently only do quasi time stepping, which simplifies a lot of things.
For actual time stepping we need further things like:

- time related matrices
- event management (contact)
- adaptivity
    - resets
    - restarts
- recycling (reuse of matrices for certain time steppers)
- transformation management for e.g., Runge-Kutta methods see e.g. Hairer and Warnner
- enforcement of boundary conditions while stepping through time
- time interpolation
- if the dimensionality of the problem is changed (think about space adaptivity which shrinks or enlarges the degrees of freedom) then, the solver needs to manage this

First try to use DiffEq with Ferrite.jl stuff can be seen [here](https://ferrite-fem.github.io/Ferrite.jl/stable/examples/ns_vs_diffeq/)

Most things are already handled by DiffEq and so we should reuse that.
@termi-official

**Comment by KnutAM** [comment]
Yes, absolutely. We can also consider for dynamics problems, the equation (for linear elasticity) is $M\ddot{u}+Ku=f(t)$ as compared to the DAE in NS example...
Currently, we are only supporting quasi-static problems and it would of course be nice to support dynamics too!
